it/src/main/java/org/corfudb/universe/node/server/docker/DockerCorfuServer.java

```diff
@@ -0,0 +1,435 @@
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+
+            if (!container.state().running() && !container.state().paused()) {
+                log.warn("The container `{}` is not running", container.name());
+                return;
+            }
+            docker.killContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't kill Corfu server: " + params.getName(), ex);
+        }
+    }
+
+    /**
+     * Immediately kill and remove the docker container
+     *
+     * @throws NodeException this exception will be thrown if the server can not be killed.
+     */
+    @Override
+    public void destroy() {
+        log.info("Destroying the Corfu server. Docker container: {}", params.getName());
+
+        if (destroyed.getAndSet(true)) {
+            log.debug("Already destroyed: {}", params.getName());
+            return;
+        }
+
+        try {
+            kill();
+        } catch (NodeException ex) {
+            log.warn("Can't kill container: {}", params.getName());
+        }
+
+        collectLogs();
+
+        try {
+            docker.removeContainer(params.getName());
+        } catch (DockerException | InterruptedException ex) {
+            throw new NodeException("Can't destroy Corfu server. Already deleted. Container: " + params.getName(), ex);
+        }
+    }
+
+    @Override
+    public Stress getStress() {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    /**
+     * Disconnect the container from docker network. This disconnects partitions from other containers.
+     * The test runtime can still connect to this. This causes a case of partial partitions.
+     *
+     * @throws NodeException this exception will be thrown if the server can not be disconnected
+     */
+    @Override
+    public void disconnect() {
+        log.info("Disconnecting the server from docker network. Docker container: {}", params.getName());
+
+        clusterParams.getNodesParams().forEach(neighbourServer -> {
+            if (neighbourServer.equals(params)){
+                return;
+            }
+
+            try {
+                String networkName = universeParams.getNetworkName();
+                IpamConfig ipamConfig = docker.inspectNetwork(networkName).ipam().config().get(0);
+                String gateway = ipamConfig.gateway();
+
+                ContainerInfo server = docker.inspectContainer(neighbourServer.getName());
+                String neighbourhoodIp = server.networkSettings().networks().values().asList().get(0).ipAddress();
+
+                if (StringUtils.isEmpty(neighbourhoodIp)){
+                    throw new NodeException("Empty ip address. Container: " + neighbourServer.getName());
+                }
+
+                // iptables -A INPUT -s $gateway -j ACCEPT
+                execCommand("iptables", "-A", "INPUT", "-s", gateway, "-j", "ACCEPT");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "iptables" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)